storybookjs/storybook
# Avoid logging an object on compilation errors

## Issue

In certain configurations Storybook might log the whole Webpack
build object when warnings or errors are thrown. This was caused
by a `logger.error` call inside the `index.js` catch.

## What I did

This fixes by logging the error only when it's an instance of an error.

## How to test

This branch was tested on https://github.com/pagarme/pilot. Before
this changes, if an error happens while compiling (e.g. imported file not
found) a big object is logged:

![screenshot from 2017-10-24 11-31-24](https://user-images.githubusercontent.com/1611639/31945774-e9de0816-b8ae-11e7-8a71-d5ae6aff2bde.png)

Resolves
#1889.

Is this testable with jest or storyshots?

Nope.

Does this need a new example in the kitchen sink apps?

Nope.

Does this need an update to the documentation?

Nope.

If your answer is yes to any of these, please make sure to include it in your PR.
